test/test_response.py

```diff
@@ -198,7 +200,9 @@ def test_io(self):
         self.assertRaises(IOError, resp3.fileno)
 
     def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+        sock = socket.socket()
+        self.addCleanup(sock.close)
```

Comment:
I’ve been able to reduce it to the following minimal case of FD leakage:

```python
import socket
import unittest

import http.client as httplib


class TestResponse(unittest.TestCase):
    def test_io_closed_consistently(self):
        hlr = httplib.HTTPResponse(socket.socket())


if __name__ == '__main__':
    unittest.main()
```

If you stick that in `urllib3/test` and run it, you get the FD leakage. If you run it in a clean environment, you don’t – seems like something in the urllib3 repo is causing us to hold on to the socket (which might explain leaks elsewhere).